packages/components/src/components/textarea/textarea.ts

```diff
@@ -247,93 +249,150 @@ export default class SdTextarea extends SolidElement implements SolidFormControl
+      disabled: 'border-neutral-500',
+      readonly: 'border-neutral-800',
+      activeInvalid: 'border-error border-2',
+      activeValid: 'border-success border-2',
+      active: 'border-primary border-2',
+      invalid: 'border-error',
+      valid: 'border-success',
+      default: 'border-neutral-800'
+    }[inputState];
 
     return html`
       <div
         part="form-control"
-        class=${classMap({
-      'form-control': true,
-      'form-control--small': this.size === 'small',
-      'form-control--medium': this.size === 'medium',
-      'form-control--large': this.size === 'large',
-      'form-control--has-label': hasLabel,
-      'form-control--has-help-text': hasHelpText
-    })}
+        class=${cx(
+          'form-control text-left',
+          hasLabel && 'form-control--has-label',
+          hasHelpText && 'form-control--has-help-text',
+          this.disabled && 'pointer-events-none cursor-not-allowed'
+        )}
       >
         <label
           part="form-control-label"
-          class="form-control__label"
+          class=${cx('form-control-label mb-2', hasLabel ? 'inline-block' : 'hidden', textSize)}
           for="input"
           aria-hidden=${hasLabel ? 'false' : 'true'}
         >
-          <slot name="label">${this.label}</slot>
+          <slot name="label">${this.label}${this.required ? '*' : ''}</slot>
         </label>
 
-        <div part="form-control-input" class="form-control-input">
+        <div part="form-control-input" class="form-control-input relative w-full">
+          <div
+            part="border"
+            class=${cx('absolute w-full h-full pointer-events-none border rounded-default', borderColor)}
+          ></div>
           <div
             part="base"
-            class=${classMap({
-      textarea: true,
-      'textarea--small': this.size === 'small',
-      'textarea--medium': this.size === 'medium',
-      'textarea--large': this.size === 'large',
-      'textarea--standard': !this.filled,
-      'textarea--filled': this.filled,
-      'textarea--disabled': this.disabled,
-      'textarea--focused': this.hasFocus,
-      'textarea--empty': !this.value,
-      'textarea--resize-none': this.resize === 'none',
-      'textarea--resize-vertical': this.resize === 'vertical',
-      'textarea--resize-auto': this.resize === 'auto'
-    })}
+            class=${cx(
+              'textarea px-4 flex flex-row items-center rounded-default textarea--resize-none',
+              this.size === 'sm' && 'textarea--small',
+              this.size === 'md' && 'textarea--medium',
+              this.size === 'lg' && 'textarea--large',
+              // Vertical Padding
+              this.size === 'lg' ? 'py-2' : 'py-1',
+              // States
+              !this.disabled && !this.readonly ? 'hover:bg-neutral-200' : '',
+              this.readonly && 'bg-neutral-100',
+              isInvalid && 'form-control-input--invalid',
+              textColor,
```

Comment:
And the same thing goes for textColor:

```
{
  disabled: 'text-neutral-500',
  readonly: 'text-black',
  activeInvalid: 'text-error',
  activeValid: 'text-success',
  active: 'text-black',
  invalid: 'text-error',
  valid: 'text-success',
  default: 'text-black'
}[inputState]
```

I personally find it more readable to resolve it directly within the class.